content/post/language_migrations.md

```diff
@@ -0,0 +1,512 @@
+
+Here, a valuable strategy which makes us win both coding and computational time, is to use [`PyEMD`](https://pypi.python.org/pypi/pyemd) : a Python wrapper for the [Earth Mover's Distance](https://en.wikipedia.org/wiki/Earth_mover's_distance) that runs with Numpy. This measure -- better than euclidean distance for histogram comparison -- is particularly beneficial because it is based on [Linear Programming](https://en.wikipedia.org/wiki/Linear_programming) (LP). Indeed, it can be seen as the solution of the following [transportation problem](http://www.me.utexas.edu/~jensen/models/network/net8.html), where \\(~\\sum_ {i=1}^N s_ i = \\sum_ {j=1}^N d_ j = 1\\)
+
+<img src="/post/language_migrations/emd.png" style="width: 500px;">
+<p align="center" class="caption">Transportation Problem with supplies and demands</p>
+
+You can see that for every pair of consecutive years, the amount of bytes are either considered as "supplies" to not exceed, for the previous year or as "demands" to satisfy, for the following one. Besides, since \\(~\\sum_ {i=1}^N s_ i = \\sum_ {j=1}^N d_ j = 1\\),
+it becomes a [cost-minimization flow problem](https://en.wikipedia.org/wiki/Minimum-cost_flow_problem) whose formulation is detailed above.
+
+\\begin{equation}
+\\arg \\min \\sum_ {i=1}^N \\sum_ {j=1}^N x_ {ij}
+\\end{equation}
+\\begin{equation}\\label{non-negative}
+x_ {ij} \\geq 0,
+\\end{equation}
+\\begin{equation}\\label{sum1}
+\\sum_ {j=1}^N x_ {ij} = s_ i,
+\\end{equation}
+\\begin{equation}\\label{sum2}
+\\sum_ {i=1}^N x_ {ij} = d_ j,
+\\end{equation}
+\\begin{equation}\\label{emd}
+\\sum_ {i=1}^N \\sum_ {j=1}^N x_ {ij} = 1
+\\end{equation}
+
+Let's make a little aside here. First, note that for an unbiased matter, all edges' "costs" are positive and set equal to one. Furthermore, to make our problem stick to classical [flow minimization formulation](https://en.wikipedia.org/wiki/Minimum-cost_flow_problem), we would have to add an artificial source and sink on both sides of our [bipartite graph](https://en.wikipedia.org/wiki/Bipartite_graph) to ensure flow conservation. However, it doesn't change anything, especially not linearity of the problem. If you want a brief insight about such a transformation, you can take look at the last slides of [Stanford's CS97SI](https://web.stanford.edu/class/cs97si/08-network-flow-problems.pdf).
+
+## Transition Matrix
+
+Here the main code to compute the transition matrix of a specific GitHub user and between two corresponding consecutive years. Note that the main function we are using which is provided by the [`PyEMD`](https://pypi.python.org/pypi/pyemd) package is [`emd_with_flow`](https://pypi.python.org/pypi/pyemd).
```

Comment:
Here **is** the core code
The main function we use is [`emd_with_flow`](https://pypi.python.org/pypi/pyemd) which is provided by the ...